PanoptesNetClient/PanoptesNetClient/Request.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Newtonsoft.Json.Linq;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace PanoptesNetClient
+{
+    public class Request : IRequest
+    {
+        private string Resource;
+        private string Id;
+        public string Endpoint { get; set; }
+        private string Query;
+
+        public Request(string resource)
+        {
+            Resource = resource;
+            Endpoint = $"api/{resource}/";
+        }
+
+        public IRequest ById(string id)
+        {
+            Id = id;
+            return this;
+        }
+
+        public IRequest WithArgs(NameValueCollection query)
+        {
+            var collection = HttpUtility.ParseQueryString(string.Empty);
+
+            foreach (var key in query.Cast<string>().Where(key => !string.IsNullOrEmpty(query[key])))
+            {
+                collection[key] = query[key];
+            }
+            Query = $"?{collection.ToString()}";
+            return this;
+        }
+
+        public async Task<JObject> GetAsync()
+        {
+            BuildEndpoint();
+            return await ApiClient.GetAsync(this);
```

Comment:
The whole flow of this is kind of weird. 

The `ApiClient` creates an `IRequest`, which makes another `IRequest` which then goes back to call a static function on the `ApiClient`. I'm not really sure how necessary the chaining is when you can just build an IRequest object and just pass it into into the ApiClient

Reply:
The `ApiClient` creates a `Request` object which follows the `IRequest` implementation. All of the functions in `Request` use `IRequest` in their method signature as that is the expected return type. I can toy around with removing the interface, but I'd like to keep the chaining implementation so the code is easier to read/write when making requests. 

Ex. `ApiClient.Type("projects").ById("5").GetAsync` rather than `ApiClient.GetAsync("projects/5")` as the latter seems like it could get convoluted when calling for different queries, types, etc. We should talk about this in person though.